### ISSUE TITLE
Levelup Issues

### ISSUE THREAD

**Issue by seiyria** [IdleLands/IdleLands#3]
Previously, people leveled up a lot and events could take you a static percentage through your level, making high levels exceptionally easy to get through. Some possible solutions:

* diminishing returns on events at higher levels (the higher you get, the less effective events are)
* turn off events for people after a certain level
* other?

**Comment by sedgwick** [comment]
I think having the percentage reduced would do a lot to slow down the leveling up process. Perhaps the percentage based events could be for a -5 to 5 percent chance, with it being much rarer for the 5's to occur.

A problem was that many hours of step XP could be lost in an instant with a high percent XP event. So, after a certain level, the frequency of the events was high enough to negate what you earned from time spent in game and it was about getting enough XP events that you could level up before a -XP event occured.

The smaller percentage events will still be great at higher levels, since % based gains would be large amounts of XP compared to the static XP events currently implemented. It also balances on its own across all levels, and for those lucky enough to get +XP, no extreme XP changes. There are already a few other parts of the game where huge character changes can happen, such as the fate pools, so I feel a common random event doesn't have to do it as well.

And since this issue is about all level up issues, Haste was another strong ability. You already mentioned a cap for it, like was implemented before. While you're still at a level where you regularly level up by stepping, haste meant some players leveled up much faster if they happened to get a lot of haste. I also believe there are some items and item modifiers that grant lots of haste.

**Comment by seiyria** [comment]
Interesting. Historically it's been split between positive and negative (making the negative stronger, but rarer and the positive weaker, but more common), so are you proposing that they should instead be the same event and weight the probabilities so that the farther out ones are harder to have happen? I think that's doable and makes sense.

I think the fate pools are an exception. They're only in one place, and it's meant to be drastic. I get the point, though.

Haste was very powerful in the old version - especially when walking with a lot of it. Not only will I be capping the amount of haste that affects a player but I'll also be limiting the haste modifiers that exist in the game so players can't get extreme amounts. It will probably be a range of -1 to 1 per item, instead of -3 to 3.

**Comment by sedgwick** [comment]
> (making the negative stronger, but rarer and the positive weaker, but more common)

I don't recall how long many of the levels took to reach, only that one time I remember trying to time it and X level took 18 hours. That's a long time when ticks happen every 6-10 seconds (I forget how long exactly), so it usually went like this:

Hit level X, slowly gain experience over a few hours from minor events and step XP.
Get a -xp event, losing 18% of a level, dropping you back to 0 xp at level X.

With how long a level might take to get and the frequency of events, a rare event might happen several times a day. As mentioned, at higher levels that can wipe out a whole day's progress in a couple of ticks.

> same event and weight the probabilities

If that's how it works easiest for you, but then many of the events might not make sense. Probably would still need to be separate events, just have the weighing done before. Otherwise you might have "Player gains -10,000xp" or Player loses +10,000xp." Though you say it's doable, I just don't know enough about the code.

**Comment by seiyria** [comment]
It's true, a negative can wipe out your xp, and the xp events are percentage-based. I lowered them from the previous game to currently +1-10% XP, and -5-15% XP (it used to be much stronger I think). I still do think it could use some tweaking, perhaps instead of making them so bountiful, I'll lower them to be +2-4% and -3-5% (which is much less extreme). 

with that, I can keep the weights the same as they are now and see how it works. For now, I'll change it to be less extreme, but I think some revisiting will have to occur for higher levels.

**Comment by sedgwick** [comment]
After about level 45, I noticed that leveling was starting to slow down again. I've been stagnant at level 50 though, for at least 3 days. The probability of losing experience is high enough that in the course of an hour (reviewing what is currently in my adventure log), I had several -xp events ranging anywhere from -9663 (3.00% of my level) to -13777 (4.28% of my level) xp. Right now I'm earning 16xp per step, so one -xp event may take anywhere from ~50 minutes to an hour and 10 minutes to recover.

However, with high enough luck, you're getting events frequently enough that step xp can never make up for the lost xp. I can't test again with low luck until I get some more items, but even after the fix with the luck typo, I've been able to come back from around 300 luck to 749 over a few days. I don't believe it's just me either though, I've noticed other players haven't been leveling up quite as fast either. Though, that's just a quick glance, I haven't been keeping a log. It seems that it's somewhere in the 20s that the switch-over happens. Low levels don't need as much to level up, so step xp and positive events still can push you over. I usually come back to the PC and check to find myself hovering near 0% to 20% of the way to the next level, and only once up to about 50%.

At present, it seems you're better off with low luck. With less frequent events, you're less likely to trigger the -xp events. It would take roughly 28 hours to reach level 51 with just step xp at 16xp per step, so the events are having a major effect on leveling.
